buster/completers/base.py

```diff
@@ -188,24 +188,39 @@ def from_dict(cls, completion_dict: dict):
 
 
 class Completer(ABC):
+    """Generic LLM-based completer. Requires a prompt and an input to produce an output."""
+
+    def __init__(
+        self,
+        completion_kwargs: dict,
+    ):
+        self.completion_kwargs = completion_kwargs
+
+    @abstractmethod
+    def complete(self, prompt: str, user_input: str) -> Completion:
+        ...
+
+
+class DocumentAnswerer:
```

Comment:
the main reason why the split is necessary is because we user a more "vanilla" completer in the `Validator`, so instead of writing the logic in 2 separate areas to parse responses from chatGPT, we have a single one, and then the `DocumentAnswerer` can have a different instance of it